lib/AST/ASTDumper.cpp

```diff
@@ -153,7 +153,11 @@ void RequirementRepr::printImpl(raw_ostream &out, bool AsWritten) const {
 }
 
 void RequirementRepr::print(raw_ostream &out) const {
-  printImpl(out, /*AsWritten=*/false);
+  StreamPrinter printer(out);
+  printImpl(printer, /*AsWritten=*/false);
+}
+void RequirementRepr::print(ASTPrinter &out) const {
+  printImpl(out, /*AsWritten=*/true);
```

Comment:
Normally we always prefer printing as-written, but even if we didn't in this case I'd still expect the two to be the same.